module-4/backend/week_four.md

```diff
@@ -5,15 +5,48 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 Note: When you're done, submit a PR. 
 
 1. What's your favorite project management tool?
+
+I like pivotal for group projects because it is easier to see who is working on what, because pivotal has columns showing work by person. I also like that pivotal has you click through the different states instead of having to remember to drag the cards.
+
+I like waffle by myself because it is simpler, and the customizable color coded labels are helpful, but I don't have a strong preference when working alone.
+
 2. Why do we create staging environments?
+
+We create staging environments because the production environment can be different from the development environment and staging environments allow us to simulate production and make sure there are no environment issues that would cause problems. 
+
 3. What are the characteristics of a good README (in your opinion)?
+
+Fully explaining the purpose of the project, how to run the project locally, a link to the hosted site if applicable, and letting people know how they can contribute. 
+
 4. What's one main improvement you're going to make to your code regarding accessibility issues?
+
+Making sure someone can navigate the page by tabbing.
+
 5. What are some basic security concerns to be aware of when building applications?
+
+Authentication failures/attacks; sql injection; man in the middle attacks
+
 6. Why is continuous integration helpful/important?
+
+Continuous integration allows you to make sure, before you merge a PR, that the PR does not contain any code that would cause your tests to fail. This is especially important in group projects when someone may be working on an outdated version of the code, and even if the test suite passes on their branch, if it is not up to date with master they could still break some tests if they merge their branch. 
```

Comment:
and if code might break on different environments or with different versions of gems, etc.